app/Support/Routing/RestRoute.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Plugin\Support\Routing;
+
+use Plugin\Support\Http\Request;
+use WP_REST_Request;
+
+class RestRoute extends Route
+{
+    /**
+     * Dispatch the request to the route
+     *
+     * @param \Plugin\Support\Http\Request $request
+     * @return void
+     */
+    public function dispatch(Request $request)
+    {
+        $this->router->events->listen('rest_api_init', function () use ($request) {
+            register_rest_route($this->namespace(), $this->parseUri($this->uri), [
+                'methods'             => $this->methods(),
+                'callback'            => $this->handle($request),
+                'permission_callback' => '__return_true',
+            ]);
+        });
+    }
```

Comment:
`$this->router->events` should be a protected property on the router instance. Think of another way to access the event dispatcher